awswrangler/distributed/ray/datasources/arrow_parquet_datasource.py

```diff
@@ -1,28 +1,341 @@
 """Ray ArrowParquetDatasource Module."""
+# pylint: disable=redefined-outer-name,import-outside-toplevel,reimported
```

Comment:
Could you please explain what will be the benefit to us of pulling this Ray code into our library? I imagine it's to get more control over the behaviour but I am trying to get a specific example like replacing the filesystem with ours for instance

Reply:
Yes, a lot of these were imports from `ray._internal` packages that might change between releases so this is mainly to ensure our code keeps working if we were to upgrade ray. 

The 2nd benefit is precisely the one you mentioned - to customise the code, for example I intend to use wrangler's `open_s3_object` to be able to pass additional arguments to s3 calls e.g. a frequent use case is to pass`requestor-pays` option which isn't possible with ray at the moment, but I intentionally left this out of this PR for now. I'll tackle this next.

Reply:
I think it would be important to separate the code that we just copy pasted from Ray vs the changes that we made ourselves. I'm worried that with this module specifically, future maintainers of this library will assume this whole module is copy pasted. I know that for some things we might not be able to fully separate the code, but I think we should at least highlight what our changes are in the module docs.

For example
> This module is copy pasted from Ray with the following changes:
> 1. We added our own partitioning to the read methods
> 2. ....

Reply:
Done